frontend/src/pages/Graphing/EventSelection/index.tsx

```diff
@@ -0,0 +1,180 @@
+	Reload = 'Reload',
+}
+
+const EVENT_TYPES: EventType[] = [
+	EventType.Track,
+	EventType.Click,
+	EventType.Navigate,
+	EventType.Reload,
+]
+
+export const EventSelection: React.FC<Props> = ({
+	initialQuery,
+	setQuery,
+	startDate,
+	endDate,
+}) => {
+	const { projectId } = useProjectId()
+	const [eventFilters, setEventFilters] = useState('')
+	const [eventType, setEventType] = useState<EventType>(EventType.Track)
+
+	const [eventName, setEventName] = useState('')
+	const [eventNameQuery, setEventNameQuery] = useState('')
+	const [debouncedEventNameQuery, setDebouncedEventNameQuery] = useState('')
+	useDebounce(
+		() => {
+			setDebouncedEventNameQuery(eventNameQuery)
+		},
+		300,
+		[eventNameQuery],
+	)
+
+	const [getKeyValues, { data }] = useGetKeyValuesLazyQuery()
+	const keyValueOptions = useMemo(() => {
+		const searchKeys =
+			_.chain(data?.key_values || [])
+				.uniq()
+				.value() ?? []
+
+		return searchKeys.map((o) => ({
+			key: o,
+			render: o,
+		}))
+	}, [data?.key_values])
+
+	// create initial event name and filters
+	useEffect(() => {
+		const { queryParts } = parseSearch(initialQuery)
+
+		let foundEventName = ''
+		const foundFilters: string[] = []
+
+		queryParts.forEach((part) => {
+			if (part.key === 'event') {
+				foundEventName = part.value
+			} else {
+				foundFilters.push(part.text)
+			}
+		})
+
+		if (EVENT_TYPES.includes(foundEventName as EventType)) {
+			setEventType(foundEventName as EventType)
+		} else {
+			setEventName(foundEventName)
+		}
+
+		setEventFilters(foundFilters.join(' '))
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [])
+
+	// recreate query when event or filters change
+	useEffect(() => {
+		const eventNameQuery = eventName === '' ? '' : `event=${eventName}`
+		const eventFiltersQuery = eventFilters === '' ? '' : `(${eventFilters})`
+
+		setQuery([eventNameQuery, eventFiltersQuery].join(' '))
+	}, [eventFilters, eventName, setQuery])
+
+	// recreate query when event or filters change
+	useEffect(() => {
+		getKeyValues({
+			variables: {
+				product_type: ProductType.Events,
+				key_name: 'event',
+				project_id: projectId,
+				date_range: {
+					start_date: moment(startDate).format(TIME_FORMAT),
+					end_date: moment(endDate).format(TIME_FORMAT),
+				},
+				query: debouncedEventNameQuery,
+				count: 25,
+			},
+		})
+	}, [debouncedEventNameQuery, endDate, getKeyValues, projectId, startDate])
+
+	// set event name for special event types
+	useEffect(() => {
+		if (eventType !== EventType.Track) {
+			setEventName(eventType)
+		}
+	}, [eventType])
+
+	return (
+		<SearchContext initialQuery={eventFilters} onSubmit={setEventFilters}>
+			<Card>
+				<Box
+					width="full"
+					display="flex"
+					flexDirection="column"
+					gap="12"
+				>
+					<LabeledRow label="Event type" name="eventType">
+						<OptionDropdown<EventType>
+							options={EVENT_TYPES}
+							selection={eventType}
+							setSelection={setEventType}
+						/>
+					</LabeledRow>
```

Comment:
just wanted to check here but i think the group by for events isn't working correctly. guessing unrelated to these UI changes though

https://www.loom.com/share/e1fdbf9514de4f31b00a0339fb5b4530

Reply:
Think we synced here - `email` is tied to the session and not the event, so its only finding events in which `email` is present in the attributes passed to the track event. We have a card here to add in the session attributes: https://linear.app/highlight/issue/HIG-4937/allow-searching-session-attributes-on-events